runtime/interpreter/value.go

```diff
@@ -6549,6 +6931,34 @@ func (v *EphemeralReferenceValue) Equal(_ *Interpreter, other Value) BoolValue {
 		v.Authorized == otherReference.Authorized
 }
 
+func (v *EphemeralReferenceValue) ConformsToDynamicType(dynamicType DynamicType) bool {
+	refType, ok := dynamicType.(*EphemeralReferenceDynamicType)
+	if !ok {
+		return false
+	}
+
+	referencedValue := v.ReferencedValue()
+	if referencedValue == nil {
+		return false
+	}
+
+	return (*referencedValue).ConformsToDynamicType(refType.InnerType())
+}
+
+func (v *EphemeralReferenceValue) conformsToSemaType(semaType sema.Type) bool {
```

Comment:
I realized in another PR that ephemeral references are a special case, in that they allow recursive object graphs, e.g. a reference could have an implicit reference to itself.

We need to handle this potential recursion here. I already implemented handling recursion for the `DynamicType()` function in https://github.com/dapperlabs/cadence-internal/commit/ffb32bcb3db13c4c8a3fc974203af0b183476ff6. 

We shouldn't do this in this PR, so it doesn't grow too large, but could you please create a follow-up issue to handle recursion in `ConformsToDynamicType` and `conformsToSemaType` (as both use `ReferencedValue`)?

Reply:
Created https://github.com/onflow/cadence/issues/777